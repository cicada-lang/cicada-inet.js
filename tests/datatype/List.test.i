require "List.i"
require "Trivial.i"

claim six_soles -- Trivial List end

define six_soles
  null sole cons sole cons sole cons
  null sole cons sole cons sole cons
  append
end

six_soles inspect run inspect
