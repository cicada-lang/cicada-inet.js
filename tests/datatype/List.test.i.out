(null₃)-value tail-(cons₈)
(sole₆)-value head-(cons₈)
(cons₈)-value tail-(cons₉)
(sole₇)-value head-(cons₉)
(cons₉)-value tail-(cons₁₀)
(sole₈)-value head-(cons₁₀)
(cons₁₀)-value rest-(append₄)
(null₄)-value tail-(cons₁₁)
(sole₉)-value head-(cons₁₁)
(cons₁₁)-value tail-(cons₁₂)
(sole₁₀)-value head-(cons₁₂)
(cons₁₂)-value tail-(cons₁₃)
(sole₁₁)-value head-(cons₁₃)
(cons₁₃)-value!target-(append₄)
(append₄)-return

(null₃)-value tail-(cons₈)
(sole₆)-value head-(cons₈)
(cons₈)-value tail-(cons₉)
(sole₇)-value head-(cons₉)
(cons₉)-value tail-(cons₁₀)
(sole₈)-value head-(cons₁₀)
(cons₁₀)-value tail-(cons₁₆)
(sole₉)-value head-(cons₁₆)
(sole₁₀)-value head-(cons₁₅)
(sole₁₁)-value head-(cons₁₄)
(cons₁₄)-tail value-(cons₁₅)
(cons₁₅)-tail value-(cons₁₆)
(cons₁₄)-value

